### PR TITLE
docs: Fix broken requests documentation links

### PR DESCRIPTION
**Pull request by mjsqu**
----
📚 Documentation preview 📚: https://meltano-sdk--2111.org.readthedocs.build/en/2111/

### CODE REVIEW

**Comment by mjsqu** [comment]
As described, prior to this change all docstrings pointing to requests.readthedocs.io pages were broken due to a reshuffling of the requests package documentation

File: singer_sdk/streams/rest.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def requests_session(self) -> requests.Session:
             The `requests.Session`_ object for HTTP requests.
 
         .. _requests.Session:
-            https://requests.readthedocs.io/en/latest/api/#request-sessions
+            https://requests.readthedocs.io/en/latest/api.html?#requests.Session
```
Comment (edgarrmondragon) [inline singer_sdk/streams/rest.py:143]: Is the `?` needed?

Comment (mjsqu) [reply singer_sdk/streams/rest.py:143]: Appears not, it must have come out of the referral from the search page on the site - I can remove it

**Review by edgarrmondragon — APPROVED** [review]
Thanks @mjsqu!

**Pull request merged**